File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int k = nums[0];
        int count = 1;
        for (int i = 1; i < nums.length; i++) {
            if (count == 0) {
                k = nums[i];
                count = 1;
            } else if (nums[i] == k) {
                count++;  
            } else {
                count--;
            }
        }
        
        return k;  
    }
}

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        Count=Counter(nums)
        max=0
        val=nums[0]
        for num,freq in Count.items():
            if freq>max:
                max=freq
                val=num
        
        return val
        
        
        